packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java

```diff
@@ -223,10 +223,11 @@ public void requestMediaPickFromMediaLibrary(MediaSelectedCallback mediaSelected
             @Override
             public void requestMediaPickFromDeviceLibrary(MediaSelectedCallback mediaSelectedCallback, Boolean allowMultipleSelection, MediaType mediaType) {
                 mMediaPickedByUserOnBlock = true;
-                mAppendsMultipleSelectedToSiblingBlocks = false;
+                // image blocks do not respect the multiple selection flag, so we set the append as siblings flag instead
+                mAppendsMultipleSelectedToSiblingBlocks = mediaType == MediaType.IMAGE && !allowMultipleSelection;
                 mMediaSelectedCallback = mediaSelectedCallback;
                 if (mediaType == MediaType.IMAGE) {
-                    mOnMediaLibraryButtonListener.onUploadPhotoButtonClicked(allowMultipleSelection);
+                    mOnMediaLibraryButtonListener.onUploadPhotoButtonClicked(true);
```

Comment:
I wonder why we don't have the `multiple` option turned on for the `Image` block. It's kinda confusing that it allows selecting multiple files but that's not being passed and instead is being set programmatically.

Do you have any insight on why we don't use that prop `multiple` in the block?

Reply:
Hi @geriux :wave: :smile:  Thanks for taking a look at this one.

> Do you have any insight on why we don't use that prop multiple in the block

Regarding why image does not use the `multiple` flag, AFAIK, it is because it didn't need it in the past, yet the flow for image block also didn't respect it when it was introduced (i.e. multiple _already uploaded_ images could be selected via the media library for image block even prior to the introduction of the `multiple` flag). As to why not utilizing it now to specify the behavior, it is currently the only way to distinguish between which block type originated the action. In particular, it allows us to distinguish between image and gallery. We could, in theory, begin respecting this flag for image block, and including the flag as a prop for the image block on the JS side, however this would have implications fo the iOS native implementation, and we'd need a new way to distinguish between which block originated the action. Also, I believe the current situation is considered somwhat temporary, as much of this should be resolved with transforms. The aim of this PR is to take a step towards that end by implementing multiple uploads for image block.

Please let me know if that answered the question.. or not :sweat_smile:

Reply:
Thanks for the detailed explanation @mkevins ! It makes sense =)

I tested this again updating from the latest changes and it's working as expected!

Reply:
Thanks for reviewing and testing! :+1: